gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -134,51 +147,65 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
     }
   }
 
-  protected void publishData(WorkUnitState workUnitState, int branchId, Set<Path> writerOutputPathsMoved)
-      throws IOException {
+  private void publishData(WorkUnitState state, int branchId, Set<Path> writerOutputPathsMoved) throws IOException {
+    publishData(state, branchId, false, writerOutputPathsMoved);
```

Comment:
This `publishData` method (with parameter `writerOutputPathsMoved`) is for publishing multiple-task data. Publishing single task data doesn't need `writerOutputPathsMoved`. The other one is for publishing single task data. I'll add some javadoc.